### DISCUSSION TITLE
How was data4mask.py created?

### DISCUSSION THREAD

**Discussion by s-w-s** [xlcnd/isbnlib#84] [Q&A] [answered]
In the file **data4mask.py** the header says that it was produced by _isbntools-dev_ in May 2021.
Clearly, the source code referenced a range file from isbn international and, possibly, files from national or language group authorities. But looking through the github repository for _isbntools_, I have not been able to find the source code that produced **data4mask.py**. Is it available? If so, can you point me to it? Thanks in advance for any help you can offer.

**Comment by xlcnd** [comment, marked as answer]
Hi!
That file was produced using ` isbntools-dev` which is **not** public available (for now, it uses proprietary tools that are not easy to replace).

**Reply by s-w-s** [reply]
Does isbntools-dev also access the national or language group files, e.g. from Bowker, that detail assignments to publishers?

**Comment by xlcnd** [comment]
Just uses [ISBN Ranges](https://www.isbn-international.org/range_file_generation).

**Reply by s-w-s** [reply]
Thank you.

**Comment by xlcnd** [comment]
If you want python code that produce those files, see pr #120.
